Documentation/Patterns.md

```diff
@@ -0,0 +1,77 @@
+# SerenityOS patterns
+
+----
+## Introduction
+
+Overtime numerous reoccurring patterns have emerged from or where adopted by
+the serenity code base. This document aims to track and describe them so they
+can be propagated further and keep the code base consistent. 
+
+----
+## Intrusive Lists
+
+Intrusive lists are common in there Kernel and in some specific cases
```

Comment:
Minor typo here, `in there Kernel` -> `in the Kernel`

Reply:
Fixed, thanks!